bvm/ballerina-core/src/main/java/org/ballerinalang/model/util/serializer/JsonSerializer.java

```diff
@@ -65,10 +66,15 @@
         bValueProvider.register(new BBooleanBValueProvider());
         bValueProvider.register(new ArrayListBValueProvider());
         bValueProvider.register(new HashSetBValueProvider());
+        bValueProvider.register(new LinkedHashSetBValueProvider());
         bValueProvider.register(new BTypeBValueProviders.BObjectTypeBValueProvider());
         bValueProvider.register(new BTypeBValueProviders.BRecordTypeBValueProvider());
         bValueProvider.register(new BTypeBValueProviders.BAnyTypeBValueProvider());
+        bValueProvider.register(new BTypeBValueProviders.BAnydataTypeBValueProvider());
         bValueProvider.register(new BTypeBValueProviders.BArrayTypeBValueProvider());
+        bValueProvider.register(new BTypeBValueProviders.BMapTypeBValueProvider());
+        bValueProvider.register(new BTypeBValueProviders.BUnionTypeBValueProvider());
+        bValueProvider.register(new BTypeBValueProviders.BAttachedFunctionBValueProvider());
```

Comment:
We need to clean up unnecessary logic from here.  Please create an issue, we can do it in the next iteration.

Reply:
Removed redundant `BUnionTypeBValueProvider` and `BAttachedFunctionBValueProvider` in 
82d9e98